### PR TITLE
FEATURE: Separate sidebar links for anniv and bday

### PR DESCRIPTION
**Pull request by oblakeerickson**
Instead of a single link titled "Cakeday" there will now be two links
added to the sidebar: "Anniversary" and "Birthday".

image

I didn't add any tests because there aren't any existing tests for sidebar links and because it is *just* using the discourse core api `api.decorateWidget`. 

Also looks like the `api.addCommunitySectionLink` api is still marked EXPERIMENTAL so I didn't use it.

### CODE REVIEW

File: assets/javascripts/discourse/initializers/cakeday.js [hunk 1]
```diff
@@ -155,21 +155,29 @@ function initializeCakeday(api) {
       }
     });
 
-    api.decorateWidget("hamburger-menu:generalLinks", () => {
-      let route;
+    if (cakedayEnabled) {
+      api.decorateWidget("hamburger-menu:generalLinks", () => {
+        let route = "cakeday.anniversaries.today";
```
Comment (keegangeorge) [inline assets/javascripts/discourse/initializers/cakeday.js:160]: I think we no longer need `let` here. Could we use `const` instead **or** simply inline it:
```javascript
return {
   route: "cakeday.anniversaries.today",
   label: "anniversaries.title",
   className: "cakeday-link",
 };
```

File: assets/javascripts/discourse/initializers/cakeday.js [hunk 2]
```diff
@@ -155,21 +155,29 @@ function initializeCakeday(api) {
...
+      api.decorateWidget("hamburger-menu:generalLinks", () => {
+        let route = "cakeday.birthdays.today";
 
-      return {
-        route,
-        label: "cakeday.title",
-        className: "cakeday-link",
-      };
-    });
+        return {
```
Comment (keegangeorge) [inline assets/javascripts/discourse/initializers/cakeday.js:174]: Same as above comment

**Review by keegangeorge — APPROVED** [review]
Approved, but two minor changes mentioned in comments before merging

File: assets/javascripts/discourse/initializers/cakeday.js [hunk 3]
```diff
@@ -155,21 +155,27 @@ function initializeCakeday(api) {
       }
     });
 
-    api.decorateWidget("hamburger-menu:generalLinks", () => {
-      let route;
+    if (cakedayEnabled) {
+      api.decorateWidget("hamburger-menu:generalLinks", () => {
+
+        return {
+          "cakeday.anniversaries.today",
```
Comment (pmusaraj) [inline assets/javascripts/discourse/initializers/cakeday.js:162]: This doesn't look right, and test failures seem relevant. This needs to be `route: "cakeday.anniversaries.today"`, I believe?

**Review by pmusaraj — CHANGES_REQUESTED** [review]
Actually, I'm surprised this even works for sidebar. I guess we've added backwards-compatibility to sidebar for `api.decorateWidget`, is that correct @tgxworld? 

We should add a test and we should use `api.addCommunitySectionLink`, despite it being marked as "experimental", it's the correct way forward for most things sidebar.

**Comment by oblakeerickson** [comment]
Ya I agree. I found that api and got it working locally. Just wanted to
test it out with the sidebar plug-in first.

And ya I can work on adding a test too.

**Review by pmusaraj — APPROVED** [review]
Looks good!

**Pull request merged**